lib/excon/ssl_socket.rb

```diff
@@ -66,16 +66,22 @@ def initialize(data = {})
       end
 
       # maintain existing API
-      certificate_data = @data[:client_cert] || @data[:certificate_path]
-      private_key_data = @data[:client_key] || @data[:private_key_path]
-      private_key_pass = @data[:client_key_pass] || @data[:private_key_pass]
+      client_cert_data = if cert_data = @data[:client_cert_data]
+                           cert_data
+                         elsif path = (@data[:client_cert] || @data[:certificate_path])
+                           File.read path
+                         end
+
+      private_key_data = if key_data = @data[:private_key_data]
```

Comment:
Cool. Are you saying we can safely remove `@data[:certificate_path]` and `@data[:private_key_path]`? If so, should we remove `@data[:private_key_pass]` as well?